all/Dockerfile

```diff
@@ -1,15 +1,30 @@
 FROM python:3.11.2-slim as build
+
 RUN set -ex \
   && apt-get -y update \
   && apt-get -y install --no-install-recommends git \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
 WORKDIR /opt/octodns
-COPY requirements.txt requirements.txt
+
+COPY requirements.txt all-requirements.txt
+
+ARG OCTODNS_FLAVOR=octodns
+
+RUN set -ex; \
+  if [ "${OCTODNS_FLAVOR}" != "octodns" ] ; \
+  then head -n1 all-requirements.txt >> requirements.txt; \
```

Comment:
Please fix indentation to make this easier to read. Each command should be on its own line.

Reply:
Sorry this was my editor messing with it and I didn't fix it. I can do that.. However, the intent of this is fine IMO  - value is not set or is `octodns` build all - if flavor is set pull the matching line of the `requirements.txt`

However, it's a little bit of magic putting it inline in the Dockerfile, and trying to make it all "one line" in this way because the bash script indenting and makes it quite fragile. I thing it would be better to pull this out to a specific bash script that is pushed into the container and passed `FLAVOR` and `requirements.txt` to output the set of requirements to be used.

Reply:
A separate script sounds like a great idea 👍